File: Data Structures & Algorithms/duplicate-integer/submission-1.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        nums.sort();
        i = 1;
        while i < len(nums):
            if nums[i] == nums[i-1]:
                return True;
            i = i + 1;   
        return False; 

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.java
class Solution {
        public int[] productExceptSelf(int[] nums) {
        int[] productWithoutSelf = new int[nums.length];

        int prefix  = 1;
        int suffix  = 1;

        for (int i=0; i<nums.length; i++){
            productWithoutSelf[i] = 1;
        }

        for (int i=0; i<nums.length; i++){
            productWithoutSelf[i] *= prefix;
            prefix *= nums[i];  
        }

        for (int j=nums.length-1; j>=0; j--){
            productWithoutSelf[j] *= suffix;
            suffix *= nums[j];
        }

        return productWithoutSelf;
    }
}  
